backend/controller/dal/pubsub.go

```diff
@@ -127,6 +127,47 @@ func (d *DAL) CompleteEventForSubscription(ctx context.Context, module, name str
 	return nil
 }
 
+// ResetSubscription resets the subscription cursor to the topic's head.
+func (d *DAL) ResetSubscription(ctx context.Context, module, name string) error {
```

Comment:
You'll need a named error return value in order for `tx.CommitOrRollback()` to work - super not obvious I know.

```suggestion
func (d *DAL) ResetSubscription(ctx context.Context, module, name string) (err error) {
```